src/main/main.ts

```diff
@@ -29,13 +40,64 @@ import {
 } from './util';
 import { SentinelDesktopService } from './SentinelDesktopService';
 
-class AppUpdater {
-  constructor() {
-    log.transports.file.level = 'info';
-    autoUpdater.logger = log;
-    autoUpdater.checkForUpdatesAndNotify();
-  }
-}
+log.initialize();
+
+// Check for updates every 10minutes by default
+// updateElectronApp({
+//   updateSource: {
+//     type: UpdateSourceType.ElectronPublicUpdateService,
+//     repo: 'cxl-garage/desktop-app-sentinel',
+//     host: 'https://update.electronjs.org',
+//   },
+//   updateInterval: '5 minutes',
+//   logger: log,
+// });
+
+autoUpdater.logger = log;
+
+setInterval(() => {
+  autoUpdater.checkForUpdates();
+}, 30000);
+
+autoUpdater.on('update-downloaded', (event: UpdateDownloadedEvent) => {
+  const releaseNotesText = Array.isArray(event.releaseNotes)
+    ? event.releaseNotes
+        .map((note) => `Version ${note.version}: ${note.note}`)
+        .join('\n')
+    : event.releaseNotes;
```

Comment:
https://github.com/electron-userland/electron-builder/blob/fa3275c05b334f59453d04551fffa24bfa558e48/packages/builder-util-runtime/src/updateInfo.ts#L65 ; this is likely safe to just assume string as long as we don't set fullChangelog=true , we can ignore the ts type warning that pops up